ambry-server/src/main/java/com/github/ambry/server/RepairRequestsSender.java

```diff
@@ -217,12 +216,30 @@ public void run() {
         }  // loop until all the partitions are clean
 
         // sleep for some time
-        Thread.sleep(SLEEP_TIME_MS);
+        if (!shutdown) {
+          Thread.sleep(SLEEP_TIME_MS);
+        }
       } catch (Throwable e) {
         metrics.repairSenderErrorHandleCount.inc();
         logger.error("RepairRequests Sender: Exception when handling request", e);
       }
     }
+
+    // shutdown the database
+    shutdownDb();
+  }
+
+  /*
+   * shutdown the database
+   */
+  public void shutdownDb() {
+    if (db instanceof AutoCloseable) {
```

Comment:
RepairRequestDb already extends from Closeable, which means it's already an instance of AutoCloseable. We don't have to check it here.